### DISCUSSION TITLE
Possible to use kubespan and a wireguard client?

### DISCUSSION THREAD

**Discussion by mtb-xt** [siderolabs/talos#9094] [General]
Hi all,

I have 3 nodes connected with kubespan (neat feature!), and now I want all nodes to have a wireguard client to connect to an external machine (a vyos router, which has public IP address, which I want to send to talos nodes with something like MetalLB BGP).

I've added wireguard interface to one of talos nodes, and configured vyos router, but I can't ping talos node from the router.
Is there anything that would prevent a wireguard client when kubespan is enabled?

**Comment by smira** [comment]
KubeSpan can be used with Wireguard given that you setup things correctly, there's no restriction on that.
